cli/enginecli/test_gofp.py

```diff
@@ -1497,23 +1485,407 @@ def test_player_can_stake_into_free_session(self):
+            )
+
+        for stage, path in zip(stages_with_rewards, paths_with_rewards):
+            reward = self.gofp.get_path_reward(session_id, stage, path)
+            self.assertEqual(reward, (ZERO_ADDRESS, 0, 0))
+
+    def test_setting_same_path_reward_twice_overwrites_first_value(self):
+        """
+        Tests administrators' ability to set rewards for the stages in a session. Also tests anyone's
+        ability to view the rewards for a given stage in a given session.
+
+        Test actions:
+        1. Create inactive session
+        2. Check path rewards have default values
+        3. Set path rewards with a duplicate path
+        7. Check that rewards were correctly set (path reward set twice should use second value)
+        """
+        payment_amount = 132
+        uri = "https://example.com/test_setting_same_path_reward_twice_overwrites_first_value.json"
+        stages = (4, 1)
+        is_active = False
+
+        self.gofp.create_session(
+            self.nft.address,
+            self.payment_token.address,
+            payment_amount,
+            is_active,
+            uri,
+            stages,
+            False,
+            {"from": self.game_master},
+        )
+
+        session_id = self.gofp.num_sessions()
+
+        for stage in range(1, len(stages) + 1):
+            for path in range(1, stages[stage - 1] + 1):
+                reward = self.gofp.get_path_reward(session_id, stage, path)
+                self.assertEqual(reward, (ZERO_ADDRESS, 0, 0))
+
+        stages_with_rewards = [1, 1, 1, 1, 2, 1]
+        paths_with_rewards = [1, 2, 3, 4, 1, 3]
+        # Reward is path number except for stage 1 path 3
+        path_rewards = [1, 2, 3, 4, 1, 5]
+
+        self.gofp.set_path_rewards(
+            session_id,
+            stages_with_rewards,
+            paths_with_rewards,
+            [self.terminus.address for _ in stages_with_rewards],
+            [self.reward_pool_id for _ in stages_with_rewards],
+            path_rewards,
+            {"from": self.game_master},
+        )
+
+        for stage in range(1, len(stages) + 1):
+            for path in range(1, stages[stage - 1] + 1):
+                if stage != 1 or path != 3:
+                    reward = self.gofp.get_path_reward(session_id, stage, path)
+                    self.assertEqual(
+                        reward,
+                        (
+                            self.terminus.address,
+                            self.reward_pool_id,
+                            path,
+                        ),
+                    )
+
+        stage_1_path_3_reward = self.gofp.get_path_reward(session_id, 1, 3)
+        self.assertEqual(
+            stage_1_path_3_reward,
+            (
+                self.terminus.address,
+                self.reward_pool_id,
+                path_rewards[len(path_rewards) - 1],
+            ),
+        )
+
+
+class TestPlayerFlow(GOFPTestCase):
+    def test_player_can_stake_and_unstake_nfts(self):
```

Comment:
Is this a new test introduced in this PR?

Reply:
No, I think it's an error with the diff editor.